packages/snaps-utils/src/manifest/manifest.ts

```diff
@@ -191,6 +192,20 @@ export async function checkManifest(
     );
   }
 
+  if (!snapFiles.svgIcon) {
+    warnings.push(
+      'No icon found in `source.location.npm.iconPath`. It is highly recommended for your Snap to have an icon.',
+    );
+  }
+
+  const iconDimensions =
+    snapFiles.svgIcon && getSvgDimensions(snapFiles.svgIcon.toString());
+  if (iconDimensions && iconDimensions.height !== iconDimensions.width) {
+    warnings.push(
+      'Please use an icon with 1:1 ratio between height and height.',
```

Comment:
What do you think of something like this?

> The icon in the Snap manifest is not square. It is recommended to use a square icon for the Snap.

Reply:
Sounds good!